### PR TITLE
doc: add ref to Ronin regexp + remove duplicate example

### PR DESCRIPTION
**Pull request by noraj**

### CODE REVIEW

File: lib/ronin/fuzzing/core_ext/string.rb [hunk 1]
```diff
@@ -161,7 +159,7 @@ def repeating(lengths,&block)
   #     p str
   #   end
   #
-  # @example Replace a `Regexp::UNIX_PATH` with {Ronin::Fuzzing#format_strings}:
+  # @example Replace a [`Regexp::UNIX_PATH`](https://ronin-rb.dev/docs/ronin-support/Regexp.html) (from [ronin-support](https://github.com/ronin-rb/ronin-support) with {Ronin::Fuzzing#format_strings}:
```
Comment (postmodern) [inline lib/ronin/fuzzing/core_ext/string.rb:162]: I think it's redundant to say `from ronin-support` if we're already linking to it.

**Review by postmodern — CHANGES_REQUESTED** [review]

File: lib/ronin/fuzzing/core_ext/string.rb (hunk 1 above)
Comment (noraj) [reply lib/ronin/fuzzing/core_ext/string.rb:162]: Maybe I can just remove the ronin-support link but keep the text? The idea is that we also can know it comes from another gem just by reading (without clicking on the link).

Comment (postmodern) [reply lib/ronin/fuzzing/core_ext/string.rb:162]: I actually would change the whole line to just `"Replace a UNIX path with every {Ronin::Fuzzing#format_strings}". In the `@param` tag it should explain how Symbols are resolved to `Ronin::Support::Text::Patterns` constants and link to https://ronin-rb.dev/docs/ronin-support/Ronin/Support/Text/Patterns.html.

**Review by postmodern — CHANGES_REQUESTED** [review]

File: lib/ronin/fuzzing/core_ext/string.rb [hunk 2]
```diff
@@ -138,7 +138,8 @@ def repeating(lengths,&block)
   # Incrementally fuzzes the String.
   #
   # @param [Hash{Regexp,String,Symbol => Enumerable,Symbol}] substitutions
-  #   Patterns and their substitutions.
+  #   Patterns and their substitutions. Symbols are resolved to
+  #   [Ronin::Support::Text::Patterns](https://ronin-rb.dev/docs/ronin-support/Ronin/Support/Text/Patterns.html).
```
Comment (postmodern) [inline lib/ronin/fuzzing/core_ext/string.rb:142]: Missing the word "constants". Maybe "Symbols are resolved to constants in ..." or "resolved to Ronin::Support::Text::Patterns constants".

**Review by postmodern — CHANGES_REQUESTED** [review]

**Review by postmodern — APPROVED** [review]

**Pull request merged**